### DISCUSSION TITLE
Can we use Playwright api endpoint Test cases without browser context in Artillery?

### DISCUSSION THREAD

**Discussion by Rahulsaini286** [artilleryio/artillery#2429] [Artillery]
Can we use Playwright api endpoint Test cases without browser context in Artillery?

something like below test cases without using the browser context?

// apiTest.js
/*const { expect } = require('@playwright/test')
async function apiTest({ request }) {
    const response = await request.get('https://reqres.in/api/users?page=2', {
        ignoreHTTPSErrors: true
    });
    expect(response.status()).toBe(200);
    const users = await response.json();
    expect(users.data.length).toBeGreaterThan(0);
}
module.exports = { apiTest }; */
 
 
 // apiTest.yml
/* config:
  target: "https://reqres.in/api/users?page=2"
   
  phases:
    - arrivalRate: 10
      duration: 5
      
// Enable the Playwright engine:
  engines:
    playwright:
 
 processor: "../tests/apiTest.js"
  beforeScenario: setUserData
  
scenarios:
  - name: 'api Test'
    engine: playwright
    flowFunction: "apiTest"
    flow: []
/*

is it possible for artillery to run the above test without browser context/launch?

**Comment by hassy** [comment]
hi @Rahulsaini286 👋 No that won't work - Artillery's Playwright integration does not support all features of the Playwright Test Runner (`@playwright/test`), only the Playwright library itself.

**Reply by Rahulsaini286** [reply]
@hassy -ohh ok, But above script gets executed and pass in my case.
 it's just that -I don't want artillery to launch a browser before the script execution.

**Reply by bernardobridge** [reply]
I guess the script works because `page.request()` is valid and does the same as `context.request()` would. But we start a page with each VU, @Rahulsaini286, and currently there's no way to suppress that, as the Artillery Playwright integration is currently designed for browser testing.
